config/config-matterwick.default.json

```diff
@@ -29,5 +29,22 @@
     "FileJSON": true,
     "FileFormat": "",
     "FileLocation": ""
+  },
+  "CWS": {
+      "DATABASE": "",
+      "CWS_PAYMENT_URL": "",
+      "CWS_PAYMENT_TOKEN": "",
+      "CWS_SITEURL": "",
+      "CWS_SMTP_USERNAME": "",
+      "CWS_SMTP_PASSWORD": "",
+      "CWS_SMTP_SERVER": "",
+      "CWS_SMTP_PORT": "",
+      "CWS_SMTP_SERVERTIMEOUT": "",
+      "CWS_SMTP_CONNECTIONSECURITY": "",
+      "CWS_EMAIL_REPLYTONAME": "",
+      "CWS_EMAIL_REPLYTOADDRESS": "",
+      "CWS_EMAIL_BCCADDRESSES": "",
+      "CWS_CLOUD_URL": "",
+      "DOCKER_HUB_CREDENTIALS": ""
```

Comment:
we already have `DockerUsername` and `DockerPassword` dont need to duplicate this

Reply:
This is for the secret used to allow the C&C cluster to pull from the private docker hub (it's the .dockerconfigjson base64 string)

Reply:
To add to this, the DOCKER_HUB_CREDENTIALS is not just a base64 string of the username and password, it's a base64encoded JSON object that includes a nested base64 encoded string of username:password, as well as a bunch of other data. Here's what the one I'm using locally looks like.
```
{
	"auths": {
		"https://index.docker.io/v1/": {
			"auth": "<base64 encoded username:password>"
		}
	},
	"HttpHeaders": {
		"User-Agent": "Docker-Client/19.03.8 (darwin)"
	},
	"credHelpers": {
		"asia.gcr.io": "gcloud",
		"eu.gcr.io": "gcloud",
		"gcr.io": "gcloud",
		"marketplace.gcr.io": "gcloud",
		"staging-k8s.gcr.io": "gcloud",
		"us.gcr.io": "gcloud"
	},
	"experimental": "disabled"
}
```
If you want me to, I can create the nested structs required for this, but I was worried it'll make the kubernetes.go file fairly bloated and messy. The way its done for existing x-cws deployments is just like I've done in the template: (check your DM's) , this will be the value of of DOCKER_HUB_CREDENTIALS in the json configuration.

Reply:
hum, ok, lets make it easier, in the `DOCKER_HUB_CREDENTIALS` i will add the dockercfg json for the user we will use.
lets keep in the way you did.